### ISSUE TITLE
InChI identifying descriptor vs. structural descriptor 

### ISSUE THREAD

**Issue by tilfischer** [semanticchemistry/semanticchemistry#57]
Following up on Stuart Chalk's issue to add RInChI and MInChI [#56](https://github.com/semanticchemistry/semanticchemistry/issues/56#issue-1440972409), I would like to raise that InChI is currently a "structural descriptor" in Cheminf, while InChI should be actually an "identifying descriptor". Notably, the InChIKey is an "identifying descriptor" in Cheminf.

It is possible to regenerate a structure from InChI, but this has some [limitations](https://www.inchi-trust.org/technical-faq-2/#16.9), as the InChI is an Identifier by design and not a structural representation such as SMILES.

I would suggest to have InChI as an "identifying descriptor" in Cheminf and to change the description to: "An InChI descriptor is a**n identifying descriptor** which conforms to the InChI format specification."

**Comment by hujo91** [comment]
I believe this discussion applies not only to ["InChI descriptor" (CHEMINF:000113)](https://semanticscience.org/resource/CHEMINF_000113), but also to ["NInChI descriptor" (CHEMINF:000517)](https://semanticscience.org/resource/CHEMINF_000517).
In my opinion, both should be better subsumed under ["identifying descriptor" (CHEMINF:000061)](https://semanticscience.org/resource/CHEMINF_000061), rather than ["structural descriptor" (CHEMINF:000085)](https://semanticscience.org/resource/CHEMINF_000085).

**Comment by egonw** [comment]
> It is possible to regenerate a structure from InChI, but this has some [limitations](https://www.inchi-trust.org/technical-faq-2/#16.9), as the InChI is an Identifier by design and not a structural representation such as SMILES.

@tilfischer, I am sorry I did not reply ealier. I think the categorization is correct, precisely because of the point you bring up: the InChI explains the structure. Yes, it has limitations, but so does the SMILES. The `chemical graph` generally has limitations in how to represents the chemical structure (think delocalisation, etc).

**Comment by tilfischer** [comment]
I agree to the limitations of a chemical graph in general, but this model works out quit nicely in general, which is the reason why it is used. Part of that structure represented in a chemical graph are atoms and bonds with their bond orders. 

However, InChI does not know bond orders but connectivities. Consequently, it does not know the whole structure but only that atoms are connected. SMILES on the other side knows bond orders and is a structural descriptor, while InChI is a identifying descriptor. Possible @gblanke02 could provide more guidance. 

Best,
Tillmann

**Comment by gblanke02** [comment]
About InChI:
InChI works with connectivity (the bond order as such is neglected) but counts all ligands including H-atoms (in the H layer) leaving from each atom. By that the bond order is replaced by a full ligand description. If you ML focus is on bonds, smiles are preferrable. But if you are interested more in the skeletons InChI may be more preferrable for ML.

Cheers,
Gerd

Von: tilfischer ***@***.***>
Gesendet: Freitag, 12. April 2024 11:16
An: semanticchemistry/semanticchemistry ***@***.***>
Cc: ***@***.***; Mention ***@***.***>
Betreff: Re: [semanticchemistry/semanticchemistry] InChI identifying descriptor vs. structural descriptor (Issue #57)


I agree to the limitations of a chemical graph in general, but this model works out quit nicely in general, which is the reason why it is used. Part of that structure represented in a chemical graph are atoms and bonds with their bond orders.

However, InChI does not know bond orders but connectivities. Consequently, it does not know the whole structure but only that atoms are connected. SMILES on the other side knows bond orders and is a structural descriptor, while InChI is a identifying descriptor. Possible @gblanke02<https://github.com/gblanke02> could provide more guidance.

Best,
Tillmann
